lib/libutee/tee_api_property.c

```diff
@@ -162,290 +176,274 @@ static TEE_Result propget_get_property(TEE_PropSetHandle h, char *name,
 			return res;
 
 		if (idx < eps_len)
-			return propget_get_ext_prop(eps + idx, pv);
+			return propget_get_ext_prop(eps + idx, type, buf, len);
 		idx -= eps_len;
 
-		size = sizeof(pv->u);
 		res = utee_get_property((unsigned long)pe->prop_set, idx,
-					0, 0, &pv->u, &size, &prop_type);
+					0, 0, buf, len, &prop_type);
 		if (res == TEE_ERROR_ITEM_NOT_FOUND)
 			res = TEE_ERROR_BAD_PARAMETERS;
 	}
 
-	if (res == TEE_SUCCESS)
-		pv->type = prop_type;
+	*type = prop_type;
 	return res;
 }
 
 TEE_Result TEE_GetPropertyAsString(TEE_PropSetHandle propsetOrEnumerator,
-				   char *name, char *valueBuffer,
-				   uint32_t *valueBufferLen)
+				   char *name, char *value,
+				   uint32_t *value_len)
 {
 	TEE_Result res;
-	struct prop_value pv;
 	size_t l;
-	size_t bufferlen;
+	enum user_ta_prop_type type;
+	void *tmp_buf = 0;
+	uint32_t tmp_len;
+	uint32_t uint32_val;
+	TEE_UUID uuid_val;
+	TEE_Identity identity_val;
 
-	if (valueBuffer == NULL || valueBufferLen == NULL) {
+	if (!value || !value_len) {
 		res = TEE_ERROR_BAD_PARAMETERS;
-		goto err;
+		goto out;
 	}
 
-	bufferlen = *valueBufferLen;
+	tmp_len = *value_len;
+	if (tmp_len < sizeof(TEE_Identity))
+		tmp_len = sizeof(TEE_Identity);
+	tmp_buf = malloc(tmp_len);
```

Comment:
I think TAs can only use `TEE_Malloc()`, and not `malloc` directly as it is not ensure it exists. However, in libutee, we can use "other" functions, like `malloc`, as we know it is implemented. This is the same for usage of `memset`, which should be `TEE_MemFill` in TAs, or `strlcpy`, which cannot be used in TAs